server/application-server/src/main/java/de/tum/in/www1/hephaestus/codereview/CodeReviewService.java

```diff
@@ -0,0 +1,212 @@
+                        }
+                        try {
+                                pullrequest.setAuthor(getActorFromGHUser(pr.getUser()));
+                        } catch (IOException e) {
+                                logger.error("Error while fetching PR author!");
+                                pullrequest.setAuthor(null);
+                        }
+
+                        return pullrequest;
+                }).collect(Collectors.toList());
+                repository.setPullRequests(prs);
+                pullrequestRepository.saveAll(prs);
+                repositoryRepository.save(repository);
+                return repository;
+        }
+
+        /**
+         * Retrieves the comments of a given pull request.
+         * 
+         * @param pr          The GH pull request.
+         * @param pullrequest Stored PR to which the comments belong.
+         * @return The comments of the given pull request.
+         * @throws IOException
+         */
+        private List<Comment> getCommentsFromGHPullRequest(GHPullRequest pr, Pullrequest pullrequest)
+                        throws IOException {
+                CommentConverter commentConverter = new CommentConverter();
+                List<Comment> comments = pr.queryComments().list().toList().stream()
+                                .map(comment -> {
+                                        Comment c = commentConverter.convert(comment);
+                                        c.setPullrequest(pullrequest);
+                                        Actor author;
+                                        try {
+                                                author = getActorFromGHUser(comment.getUser());
+                                                author.addComment(c);
+                                                author.addPullrequest(pullrequest);
+                                        } catch (IOException e) {
+                                                logger.error("Error while fetching author!");
+                                                author = null;
+                                        }
+                                        c.setAuthor(author);
+                                        return c;
+                                }).collect(Collectors.toList());
+                return comments;
+        }
+
+        private Actor getActorFromGHUser(GHUser user) {
+                Actor actor = actorRepository.findByLogin(user.getLogin());
+                if (actor == null) {
+                        actor = new ActorConverter().convert(user);
+                        actorRepository.save(actor);
+                }
+                return actor;
+        }
+
+        /**
+         * GraphQL implementation of fetching the hephaestus Github repository.
+         * 
+         * @see #fetchRepository(String)
+         * @return The hephaestus repository.
+         */
+        public Repository getHephaestusRepository() {
```

Comment:
I think that could be removed, but I think you can keep the GraphQL client initialization as we might want to use it in the future. But I think we can also remove it completely for now until it is necessary again. Otherwise we would have unused code that we might also need to maintain.